Sources/Algorithms/Stride.swift

```diff
@@ -0,0 +1,112 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// stride(by:)
+//===----------------------------------------------------------------------===//
+
+extension Collection {
+  /// - Complexity: O(_1_) and access to the next value in the stride
+  /// is O(_1_) if the collection conforms to `RandomAccessCollection`,
+  /// otherwise O(_k_) where `k` is the stride `step`.
+  ///
+  /// - Parameter step: The amount to step with each iteration.
+  /// - Returns: Returns a collection stepping through the elements by the
+  /// specified amount.
+  public func stride(by step: Int) -> Stride<Self> {
```

Comment:
I wonder if this would best be `striding` to fit the ed/ing rule for naming nonmutating members. That said, I also worry about ambiguity with the existing `StrideTo`/`StrideThrough` types: are there other terms that are often used for this operation?

Reply:
I am in agreement that `striding` is probably a better fit for this name for the reasons you've mentioned. I'm a little unsure that renaming the type to something else would be a good idea as the notion of a stride is widely known and something which is obvious as soon as you see it. In one of my earlier implementations I had named the type `StridingCollection` which I would not be against being explicit about, I removed the suffix `Collection` to keep in step with the rest of the algorithms in this repository. It would be good to see what other feedback I get regarding the naming of this before I change it. Thanks for your thoughts @xwu